test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,468 @@
+        assertNotNull(reader0.readNextEvent(500).getEvent());
+        assertNotNull(reader0.readNextEvent(500).getEvent());
+        assertNotNull(reader0.readNextEvent(500).getEvent());
+        assertNull(reader0.readNextEvent(500).getEvent());
+        reader0.close();
+
+        ArrayList<SegmentRange> segmentList1 = Lists.newArrayList(batchClient.getSegments(stream, StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        log.info("Segment List1 :{}", segmentList1);
+
+        Map<Segment, Long> map = segmentList1.stream().collect(Collectors.toMap(SegmentRange::getSegment, value -> value.getEndOffset()));
+
+        assertTrue(nextStreamCut5 != null);
+        assertTrue(nextStreamCut5.asImpl().getPositions().size() == 2);
+        assertTrue(nextStreamCut5.asImpl().getPositions().containsKey(segment1) &&
+                nextStreamCut5.asImpl().getPositions().containsKey(segment2));
+        assertTrue(map.get(segment1).longValue() <= nextStreamCut5.asImpl().getPositions().get(segment1).longValue());
+        assertTrue(map.get(segment2).longValue() <= nextStreamCut5.asImpl().getPositions().get(segment2).longValue());
+        //Scaling up end
+    }
+
+    @Test(timeout = 60000)
+    public void getNextStreamCutWithScaleDownTest() throws SegmentTruncatedException, ExecutionException, InterruptedException {
+        String streamName = "testStreamSegmentScaleDown";
+        String streamScope = "testScopeSegmentScaleDown";
+        String readerGroupName = "testReaderGroupSegmentScaleDown";
+        String readerName = UUID.randomUUID().toString();
+
+        StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+        ClientConfig clientConfig = ClientConfig.builder()
+                .controllerURI(fetchControllerURI())
+                .build();
+        URI controllerURI = clientConfig.getControllerURI();
+
+        Stream stream = Stream.of(streamScope, streamName);
+
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(streamScope, controller, connectionFactory);
+
+        assertTrue(controller.createScope(streamScope).join());
+        assertTrue(controller.createStream(streamScope, streamName, config).join());
+
+        @Cleanup
+        BatchClientFactory batchClient = BatchClientFactory.withScope(streamScope, clientConfig);
+
+        List<SegmentRange> ranges = Lists.newArrayList(batchClient.getSegments(stream, StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        List<Segment> list = ranges.stream().map(SegmentRange::getSegment).collect(Collectors.toList());
+        assertEquals(1, list.size());
+
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+
+        // write events to stream 30 * 5  = 150 bytes
+        write30ByteEvents(5, writer);
```

Comment:
Please, make the number of events to be written a variable that you can use later on to reason about the number of events to read in a loop.